platforms/core-configuration/input-tracking/src/main/java/org/gradle/internal/configuration/inputs/AccessTrackingProperties.java

```diff
@@ -237,6 +238,7 @@ public boolean replace(Object key, Object expectedOldValue, Object newValue) {
     }
 
     @Override
+    @Nullable
     public Object replace(Object key, Object value) {
```

Comment:
💅 This will be the preferable way of null-annotating the return type of method. Could you update this and other places like this?

See this comment for reference: https://github.com/gradle/gradle/pull/26677#discussion_r1362451380

```suggestion
    public @Nullable Object replace(Object key, Object value) {
```

Reply:
>This will be the preferable way of null-annotating the return type of method. Could you update this and other places like this?

The preference there comes from a false assumption that `TYPE_USE` annotation requires placing next to a type. [JLS](https://docs.oracle.com/javase/specs/jls/se11/html/jls-9.html#jls-9.7.4) states that the position of the annotation relative to method modifiers doesn't really matter. What gets annotated is determined by the annotation's `Target`- type, method, or both:
>The grammar of the Java programming language unambiguously treats annotations at these locations as modifiers for a declaration ([§8.3](https://docs.oracle.com/javase/specs/jls/se11/html/jls-8.html#jls-8.3)), but that is purely a syntactic matter. Whether an annotation applies to the declaration or to the type of the declared entity - and thus, whether the annotation is a declaration annotation or a type annotation - depends on the applicability of the annotation's type:
...
> * If the annotation's type is applicable in type contexts, and not in the declaration context corresponding to the declaration, then the annotation is deemed to apply only to the type which is closest to the annotation.

Given that the vast majority of code applies annotation on a separate line, I'd rather follow this convention for now.

Reply:
Ah, I wasn't aware that it works there. I feel like there is still some issue with formatting, but I can't reproduce in IntelliJ now so that might have been a bug that is now fixed.